File: mysql_examples/recipes/apache/mysql_status/mysql_status.py
#!/usr/bin/python
# mysql_status.py - print some MySQL server status information

import MySQLdb
import Cookbook

# Print header, blank line, and initial part of page

print """Content-Type: text/html

<html>
<head>
<title>MySQL Server Status</title>
</head>
<body bgcolor="white">
"""

# Connect to database
conn = Cookbook.connect ()

# Retrieve status information
cursor = conn.cursor ()
cursor.execute ("SELECT NOW(), VERSION()")
(now, version) = cursor.fetchone ()
# SHOW STATUS variable values are in second result column
cursor.execute ("SHOW /*!50002 GLOBAL */ STATUS LIKE 'Questions'")
queries = cursor.fetchone ()[1]
cursor.execute ("SHOW /*!50002 GLOBAL */ STATUS LIKE 'Uptime'")
uptime = cursor.fetchone ()[1]
q_per_sec = "%0.2f" % (float (queries) / float (uptime))
cursor.close ()

# Disconnect from database
conn.close ()

# Display status report
print "<p>Current time: " + str(now) + "</p>"
print "<p>Server version: " + version + "</p>"
print "<p>Server uptime (seconds): " + uptime + "</p>"
print "<p>Queries processed: " + queries + " (" \
    + q_per_sec + " queries/second)</p>"

# Print page trailer
print """
</body>
</html>
"""

File: mysql_examples/recipes/apache/mysql_status/mysql_status.php
<?php
# mysql_status.php - print some MySQL server status information

require_once "Cookbook.php";

# Helper function: execute a query, return first result row as an array

function fetch_row ($conn, $stmt)
{
  $result =& $conn->query ($stmt);
  if (PEAR::isError ($result))
    die ("Cannot execute query: "
         . htmlspecialchars ($result->getMessage ()));
  $row = $result->fetchRow ();
  $result->free ();
  return ($row);
}
?>

<html>
<head>
<title>MySQL Server Status</title>
</head>
<body bgcolor="white">

<?php
# Connect to database
$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
  die ("Cannot connect to server: "
       . htmlspecialchars ($conn->getMessage ()));

# Retrieve status information
list ($now, $version) = fetch_row ($conn, "SELECT NOW(), VERSION()");
# SHOW STATUS variable values are in second result column
list ($junk, $queries) =
  fetch_row ($conn, "SHOW /*!50002 GLOBAL */ STATUS LIKE 'Questions'");
list ($junk, $uptime) =
  fetch_row ($conn, "SHOW /*!50002 GLOBAL */ STATUS LIKE 'Uptime'");
$q_per_sec = sprintf ("%.2f", $queries/$uptime);

# Disconnect from database
$conn->disconnect ();
?>

<!-- Display status report -->
<p>Current time: <?php print ($now); ?></p>
<p>Server version: <?php print ($version); ?></p>
<p>Server uptime (seconds): <?php print ($uptime); ?></p>
<!-- #@ _QUERIES_ -->
<p>Queries processed:  <?php print ($queries); ?>
 (<?php print ($q_per_sec); ?> queries/second)</p>
<!-- #@ _QUERIES_ -->

</body>
</html>

